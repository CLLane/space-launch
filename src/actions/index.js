export const nextLaunchData = (launch) => ({
  type: 'NEXT_LAUNCH',
  launch,
})

export const monthLaunchData = (launches) => ({
  type: 'MONTH_LAUNCH',
  launches
})

export const rocketTypeData = (rockets) => ({
  type: 'ROCKET_TYPE',
  rockets
})

export const getMissionData = (missions) => ({
  type: 'MISSION_DATA',
  missions
})

export const getCompanyData = (companies) => ({
  type: 'COMPANY_DATA',
  companies
})