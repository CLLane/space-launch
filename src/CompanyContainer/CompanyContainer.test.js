import React from "react";
import { shallow } from "enzyme";
import {
  CompanyContainer,
  mapDispatchToProps,
  mapStateToProps
} from "./CompanyContainer";
import { getCompanyData } from "../actions/index";

describe("CompanyContainer", () => {
  let wrapper;
  let mockGetCompanyData = jest.fn();
  let mockCompanies = [{ name: "space x" }];
  let mockCompany = {
    id: 1,
    name: "Chris"
  };

  beforeEach(() => {
    wrapper = shallow(
      <CompanyContainer
        company={mockCompany}
        getCompanyData={mockGetCompanyData}
        companies={mockCompanies}
      />
    );
  });

  it("should match a snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a companies array', () => {
    const mockState = {
      companies: [{name: 'space x'}]
    }
    const expected = {
      companies: [{ name: 'space x' }]
    }
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch with a getCompanyData action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = getCompanyData(['mock']);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getCompanyData(['mock'])

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
