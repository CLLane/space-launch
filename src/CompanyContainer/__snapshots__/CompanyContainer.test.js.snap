// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyContainer should match a snapshot 1`] = `
<Fragment>
  <CompanyCards
    company={
      Object {
        "name": "space x",
      }
    }
    key="space x"
  />
</Fragment>
`;
